pkg/cloudprovider/providers/vsphere/nodemanager.go

```diff
@@ -443,3 +482,27 @@ func (nm *NodeManager) UpdateCredentialManager(credentialManager *SecretCredenti
 	defer nm.credentialManagerLock.Unlock()
 	nm.credentialManager = credentialManager
 }
+
+func (nm *NodeManager) GetHostsInZone(ctx context.Context, zone cloudprovider.Zone) ([]*object.HostSystem, error) {
+	klog.V(7).Infof("GetHostsInZone called with registeredNodes: %v", nm.registeredNodes)
+	nodeDetails, err := nm.GetNodeDetails()
+	if err != nil {
+		return nil, err
+	}
+	klog.V(4).Infof("Node Details: %v", nodeDetails)
+	// Return those hosts that are in given zone.
+	hosts := make([]*object.HostSystem, 0)
+	for _, n := range nodeDetails {
+		// Match the provided zone with the node's zone, if not empty.
+		if (zone.FailureDomain == "" || zone.FailureDomain == n.Zone.FailureDomain) && (zone.Region == "" || zone.Region == n.Zone.Region) {
```

Comment:
This compound logic expression should be tested (unit or e2e).

Reply:
I have simplified this expression now as per the way this method is consumed. Further, the tests pertaining to this PR are covered as part of https://github.com/kubernetes/kubernetes/pull/72732